packages/hash/api/src/graph/ontology/primitive/data-type.ts

```diff
@@ -53,19 +52,11 @@ export const createDataType: ImpureGraphFunction<
   });
   const schema = { $id: dataTypeUri, ...params.schema };
 
-  const { data: metadata } = await graphApi
-    .createDataType({
-      schema,
-      ownedById,
-      actorId,
-    })
-    .catch((err: AxiosError) => {
-      throw new Error(
-        err.response?.status === 409
-          ? `data type with the same URI already exists. [URI=${schema.$id}]`
-          : `[${err.code}] couldn't create data type: ${err.response?.data}.`,
-      );
```

Comment:
@hashintel/backend I've removed these custom error messages, as the FE should now get the Graph API error messages (instead of just getting a status code in the error message).

Reply:
The `code` has changed from `CREATION_ERROR` to `INTERNAL_SERVER_ERROR` so we have lost a bit of specific information here.

The `ApolloServer` constructor takes a [formatError](https://www.apollographql.com/docs/apollo-server/api/apollo-server/#formaterror) option we could use to rewrite the code as needed. 

Maybe the Graph API could send codes such as `CREATION_ERROR` which could then be passed through? I see that we were previously adding this in the Node API, but it probably makes sense for the Graph API to do it. I'm assuming that this would be part of the wider error handling strategy work so probably not for this PR, unless the Graph API is already sending a `code` along with its errors which we could start passing on to construct the ApolloError.

Reply:
> The code has changed from CREATION_ERROR to INTERNAL_SERVER_ERROR so we have lost a bit of specific information here.

That's technically true, but I've been assuming that the addition of the `CREATION_ERROR` code was not something that was particularly well thought through something that I can remove as part of this. The reason I thought this is because it's not being used in the FE, and because it's doesn't actually convey any more meaning than `INTERNAL_SERVER_ERROR` because the caller already knows the error was thrown in a `createX` function. But perhaps I've misunderstood this?

> Maybe the Graph API could send codes such as CREATION_ERROR which could then be passed through? 

I think that would be a more appropriate place for us to be defining different error codes.

> I'm assuming that this would be part of the wider error handling strategy work so probably not for this PR, unless the Graph API is already sending a code along with its errors which we could start passing on to construct the ApolloError.

Agreed, I think there's still a discussion that should be had about what the Graph API should be passing to the FE. Perhaps I should create a follow-up task so we can discuss it in BE refinement in the new year?

Reply:
I think this is covered under the wider 'error handling strategy' [task](https://app.asana.com/0/1201095311341924/1202574350052904/f) (internal) which I've teed up for discussion in Jan.